public/js/utils/source-documents.js

```diff
@@ -0,0 +1,24 @@
+let sourceDocs = {};
```

Comment:
Yeah, I don't think this belongs in redux, as it's just an optimization for the editor (if we replayed actions, it would make no difference if these docs existed or not). This is also only ever used by the editor, as far as I know.

Since it is only ever used by the editor, I think it's worth trying just keep it local to the editor. The only problem is clearing it on navigation, but that might be solved by not keeping a hard reference to the documents. I think it is easiest to explain with another commit so I'll pitch you some ideas with some actual code.